test/headers-as-array.js

```diff
@@ -25,6 +25,69 @@ test('handle headers as array', (t) => {
   })
 })
 
+test('handle multi-valued headers as array', (t) => {
+  t.plan(1)
+  const headers = ['a', '1', 'b', '2', 'c', '3', 'd', '4', 'd', '5']
+
+  const server = createServer((req, res) => {
+    t.match(req.headers, { a: '1', b: '2', c: '3', d: '4, 5' })
+    res.end()
+  })
+  t.teardown(server.close.bind(server))
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`)
+    t.teardown(client.destroy.bind(client))
+
+    client.request({
+      path: '/',
+      method: 'GET',
+      headers
+    }, () => {})
+  })
+})
+
+test('handle headers with array', (t) => {
+  t.plan(1)
+  const headers = { a: '1', b: '2', c: '3', d: ['4'] }
+
+  const server = createServer((req, res) => {
+    t.match(req.headers, { a: '1', b: '2', c: '3', d: '4' })
+    res.end()
+  })
+  t.teardown(server.close.bind(server))
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`)
+    t.teardown(client.destroy.bind(client))
+
+    client.request({
+      path: '/',
+      method: 'GET',
+      headers
+    }, () => {})
+  })
+})
+
+test('handle multi-valued headers', (t) => {
+  t.plan(1)
+  const headers = { a: '1', b: '2', c: '3', d: ['4', '5'] }
+
+  const server = createServer((req, res) => {
+    t.match(req.headers, { a: '1', b: '2', c: '3', d: '4, 5' })
```

Comment:
One thing to note: this behaves differently from fetch. Fetch's webidl checks will stringify the array, giving us `4,5` (no space), but node's http parser joins them with `, `, giving us `4, 5` (with space).

Reply:
Is this something you think needs to be fixed / changed?

Reply:
No, just wanted to point it out since it might have been missed in the original review. 👍